### PR TITLE
feat(nuxt): return `status` from `useAsyncData`

### LINKED ISSUE #18810: `status` property for `useAsyncData`/`useFetch` (+ lazy equivalents)
### Describe the feature

Follow-up of https://github.com/nuxt/nuxt/issues/15714

This feature request should provide a more fine-grained check of the status of `useFetch`/`useAsyncData` (and the lazy equivalents) by introducing a `status` ref that contains the status a a string.

## `Status` values

* `IDLE` -> Request did not start yet, e.g. because:
	* `{ immediate: false }` and `execute` was not called ye.
	* `{ server: false }` and status is logged during SSR
* `PENDING` -> Requesting data did start
* `SUCCESS` -> Requesting data succeeded
* `ERROR` -> Requesting data failed.

**Any other kinds of `status`es needed?**

### Additional information

- [X] Would you be willing to help implement this feature?
- [ ] Could this feature be implemented as a module?

### Final checks

- [X] Read the [contribution guide](https://nuxt.com/docs/community/contribution).
- [X] Check existing [discussions](https://github.com/nuxt/nuxt/discussions) and [issues](https://github.com/nuxt/nuxt/issues).

### PR DESCRIPTION
**Pull request by nicolaspayot**
### 🔗 Linked issue

resolves https://github.com/nuxt/nuxt/issues/18810

### ❓ Type of change

- [ ] 📖 Documentation (updates to the documentation, readme or JSdoc annotations)
- [ ] 🐞 Bug fix (a non-breaking change that fixes an issue)
- [ ] 👌 Enhancement (improving an existing functionality like performance)
- [x] ✨ New feature (a non-breaking change that adds functionality)
- [ ] 🧹 Chore (updates to the build process or auxiliary tools and libraries)
- [ ] ⚠️ Breaking change (fix or feature that would cause existing functionality to change)

### 📚 Description

- This PR introduces a new `status` property in the object returned by the composables `useAsyncData` / `useFetch` and their `lazy` variations.
- This `status` property is a  `ref<string>` that can contain the following values:
    - `idle` -> when request did not start
        - (`{ immediate: false }` and `execute` not called yet
        - `{ server: false }` -> `status` should be `idle` on server side
    - `pending` -> when started
    - `success` -> when request succeeded
    - `error` -> when request failed
- It aims to bring a more powerful way to track the status of an async data request. Relying on `pending` flag only has a limitation detailed in [this issue](https://github.com/nuxt/nuxt/issues/15714).
- Resolves https://github.com/nuxt/nuxt/issues/18810

### 📝 Checklist

- [x] I have linked an issue or discussion.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: docs/3.api/1.composables/use-async-data.md [hunk 1]
```diff
@@ -34,11 +34,14 @@ type AsyncData<DataT, ErrorT> = {
...
   execute: (opts?: AsyncDataExecuteOptions) => Promise<void>
   error: Ref<ErrorT | null>
+  status: Ref<AsyncDataRequestStatus>
 };
 
 interface AsyncDataExecuteOptions {
   dedupe?: boolean
 }
+
+type AsyncDataRequestStatus = 'IDLE' | 'PENDING' | 'SUCCESS' | 'ERROR'
```
Comment (danielroe) [inline docs/3.api/1.composables/use-async-data.md:44]: would you mind making these lowercase?

**Review by danielroe — COMMENTED** [review]
This is beautifully done - thank you ❤️

File: docs/3.api/1.composables/use-async-data.md (hunk 1 above)
Comment (danielroe) [reply docs/3.api/1.composables/use-async-data.md:44]: Just for consistency - our other string literal types are lowercase.

Comment (nicolaspayot) [reply docs/3.api/1.composables/use-async-data.md:44]: Makes sense, thanks.

File: packages/nuxt/src/app/composables/asyncData.ts [hunk 2]
```diff
@@ -222,6 +229,7 @@ export function useAsyncData<
     if (fetchOnServer && nuxt.isHydrating && hasCachedData()) {
       // 1. Hydration (server: true): no fetch
       asyncData.pending.value = false
+      asyncData.status.value = asyncData.error.value ? 'error' : 'success'
```
Comment (danielroe) [inline packages/nuxt/src/app/composables/asyncData.ts:232]: would it make sense instead to initialise it to this value above on L132?

Comment (danielroe) [reply packages/nuxt/src/app/composables/asyncData.ts:232]: Just to explain, I wasn't thinking we should reset to idle but rather could directly initialise it to 'error' or 'success' based on the value in the payload _rather_ than initialising it to 'idle'.

But on reflection, I think this is good too 👍

**Review by danielroe — APPROVED** [review]

File: packages/nuxt/src/app/composables/asyncData.ts (hunk 2 above)
Comment (nicolaspayot) [reply packages/nuxt/src/app/composables/asyncData.ts:232]: I misunderstood, gotcha! 👍

**Pull request merged**